### PR TITLE
add egress ips support to anonymizer

### PR DESCRIPTION
**Pull request by Sergey1011010**
## Categories

- [ ] Bugfix
- [X] Enhancement
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

No changes

## Documentation

No changes

## Unit Tests

No changes

## Privacy

No new data was collected

## Breaking Changes

Not really

## References

https://issues.redhat.com/browse/CCXDEV-4864
https://bugzilla.redhat.com/show_bug.cgi?id=???
https://access.redhat.com/solutions/???

### CODE REVIEW

File: pkg/anonymization/anonymizer.go [hunk 1]
```diff
@@ -128,6 +132,19 @@ func NewAnonymizerFromConfigClient(
...
 	}
 
+	// egress subnets
+
+	hostSubnets, err := networkClient.HostSubnets().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, hostSubnet := range hostSubnets.Items {
```
Comment (rluders) [inline pkg/anonymization/anonymizer.go:142]: I'm pretty sure that the lint will complain about these two `for`, 'cause it will say that you should use `index` or `pointer`.

**Review by rluders — CHANGES_REQUESTED** [review]

File: pkg/anonymization/anonymizer.go (hunk 1 above)
Comment (Sergey1011010) [reply pkg/anonymization/anonymizer.go:142]: Um, why?

Comment (rluders) [reply pkg/anonymization/anonymizer.go:142]: Check these links:
https://go-critic.github.io/overview#rangeValCopy-ref
https://stackoverflow.com/questions/20185511/range-references-instead-values

**Comment by tremes** [comment]
@Sergey1011010 would it be possible to test it (in our unit tests) please?

**Comment by Sergey1011010** [comment]
@tremes added the test

File: pkg/anonymization/anonymizer.go [hunk 2]
```diff
@@ -83,7 +84,7 @@ type ConfigProvider interface {
 
 // NewAnonymizer creates a new instance of anonymizer with a provided config observer and sensitive data
 func NewAnonymizer(clusterBaseDomain string, networks []string, secretsClient corev1client.SecretInterface) (*Anonymizer, error) {
-	networks = append(networks, "127.0.0.1/8")
+	networks = append(networks, "127.0.0.0/8")
```
Comment (tremes) [inline pkg/anonymization/anonymizer.go:87]: Why this change please? It's not clear to me at the first glance. We need to update also in the package doc - https://github.com/openshift/insights-operator/pull/446/files#diff-9180eb430c735bed4f79b7892293543fe58bd8bc57493f4f823ca84faa073f8eR12

Comment (Sergey1011010) [reply pkg/anonymization/anonymizer.go:87]: It was a mistake in the original code, not really a problem cuz anonymizations would still work correctly, but /8 means everything after the first byte(127) should be zeros so that it's a valid mask.

Comment (tremes) [reply pkg/anonymization/anonymizer.go:87]: Ahh I see. Thx

**Comment by tremes** [comment]
Reviewed & tested. Please remove BZ Id and I also think it would be nice to backport it to 4.8. Thanks!!
/lgtm
